src/magic_folder/snapshot.py

```diff
@@ -216,35 +216,39 @@ def create_author_from_json(data):
     return create_author(data["name"], verify_key)
 
 
-def sign_snapshot(local_author, snapshot, content_capability):
+def sign_snapshot(local_author, snapshot, content_capability, metadata_capability):
     """
     Signs the given snapshot with provided author
 
     :param LocalAuthor local_author: the author to sign the data with
 
     :param LocalSnapshot snapshot: snapshot to sign
 
-    :param str content_capability: the Tahoe immutable
+    :param bytes content_capability: the Tahoe immutable
         capability-string of the actual snapshot data.
 
+    :param bytes metadata capability: the Tahoe immutable
+        capability-string of the metadata (which is serialized JSON)
+
     :returns: bytes representing the signature (or exception on
         error).
     """
     # XXX See
     # https://github.com/LeastAuthority/magic-folder/issues/190 as
-    # this is likely insufficient (and hasn't been looked at by our
-    # cryptograhphers yet).
+    # this hasn't been looked at by our cryptograhphers yet).
     data_to_sign = (
         u"{content_capability}\n"
+        u"{metadata_capability}\n"
         u"{name}\n"
     ).format(
         content_capability=content_capability,
+        metadata_capability=metadata_capability,
         name=snapshot.name,
```

Comment:
This is more of our path confusion.  I *don't* think `snapshot.name` will be mangled in the long run.  It will be a normal relative path.  I think it is *sometimes* the mangled path currently, but probably the wrong mangled path (absolute instead of relative).  This is a separate issue that I'm not suggesting be addressed by this PR, just pointing it out.

Maybe the unmangled path is a better thing to sign anyway?  The mangling is sort of a protocol/storage detail that it might be nice to be able to change.  The *real* relative file name is part of the user data and we shouldn't want to (or be able to) change it without breaking signatures, I think.

Reply:
I don't think it really matters either way here (of course, we need to consistently use one). Whatever "name" happens to be, we will sign it ... and only accept one on the other side that's got a valid signature. How that is interpreted by the rest of the system doesn't (shouldn't!) matter here.

It does seem *slightly* better, maybe, to sign the "real actual relative path" .. but also if the name mangling/de-mangling is consistent, does it matter? If "sign the relative path" is best we should just make sure that `name` refers to that I think so this code doesn't have to "know" anything about what's in the name.